components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -881,9 +881,11 @@ public static String getQueryStringWithFrameworkContextId(String originalQuerySt
 
     public static Map<String, AuthenticatorConfig> getAuthenticatedStepIdPs(StepConfig stepConfig,
                                                                             Map<String, AuthenticatedIdPData> authenticatedIdPs) {
-
+        StringBuilder logText = null;
         if (log.isDebugEnabled()) {
-            log.debug("Finding already authenticated IdPs of the Step");
+            logText = new StringBuilder();
```

Comment:
Why this magic ?
The problem is the log will not be printed if an exception happens or someone put a return statement in the middle. so this is fragile.